src/main/java/com/example/bank/bankTransfer/transfer/history/export/XlsxTransferHistoryGenerator.java

```diff
@@ -0,0 +1,150 @@
+        autoSizeColumns(sheet);
+        createDataRows(workbook, numberOfTransferHistories, accountNumber);
+        saveWorkbookToFile(workbook, accountNumber);
+    }
+
+    private void createHeaderRow(Sheet sheet) {
+        Row headerRow = sheet.createRow(0);
+        CellStyle headerStyle = createStyleForHeader(sheet.getWorkbook());
+        IntStream.range(0, HEADER_CELL_TITLES.size()).forEach(columnIndex -> {
+            Cell cell = headerRow.createCell(columnIndex);
+            cell.setCellValue(HEADER_CELL_TITLES.get(columnIndex));
+            cell.setCellStyle(headerStyle);
+        });
+    }
+
+    private void autoSizeColumns(Sheet sheet) {
+        IntStream.range(0, HEADER_CELL_TITLES.size()).forEach(columnIndex -> {
+            sheet.autoSizeColumn(columnIndex);
+            sheet.setColumnWidth(columnIndex, sheet.getColumnWidth(columnIndex) + HORIZONTAL_PADDING);
+        });
+    }
+
+    private void createDataRows(Workbook workbook, int rowCount, String accountNumber) {
+        CellStyle dateTimeCellStyle = createDateTimeCellStyle(workbook);
+        CellStyle decimalCellStyle = createDecimalCellStyle(workbook);
+        CellStyle defaultWrapTextStyle = createDefaultWrapTextStyle(workbook);
+        List<TransferHistory> transferHistories = transferHistoryService.transferHistoryForAccountNumber(accountNumber);
+        Sheet sheet = workbook.getSheet(TRANSFER_HISTORY_SHEET_NAME);
+        for (int i = 0; i < rowCount; i++) {
+            Row row = sheet.createRow(i + 1);
+            TransferHistory history = transferHistories.get(i);
+            fillRowWithData(row, history, dateTimeCellStyle, decimalCellStyle, defaultWrapTextStyle);
+        }
+    }
+
+    private void fillRowWithData(Row row, TransferHistory history, CellStyle dateTimeCellStyle,
+                                 CellStyle decimalCellStyle, CellStyle defaultWrapTextStyle) {
+        Cell cell = null;
+        List<Object> methodsForInsertingData = createMethodsForInsertingDataIntoCells(history);
+        for (int i = 0; i < methodsForInsertingData.size(); i++) {
+            cell = row.createCell(i);
+            if (methodsForInsertingData.get(i) instanceof LocalDateTime dateTime) {
+                cell.setCellValue(dateTime);
+                cell.setCellStyle(dateTimeCellStyle);
+                row.getSheet().setColumnWidth(i, DATE_COLUMN_WIDTH);
+            } else if (methodsForInsertingData.get(i) instanceof Long number) {
+                cell.setCellValue(number);
+            } else if (methodsForInsertingData.get(i) instanceof String name) {
+                cell.setCellValue(name);
+                cell.setCellStyle(defaultWrapTextStyle);
+            } else if (methodsForInsertingData.get(i) instanceof BigDecimal bigDecimal) {
+                cell.setCellValue(bigDecimal.doubleValue());
+                cell.setCellStyle(decimalCellStyle);
+            }
+        }
+    }
```

Comment:
to wyglada jak jawny duplikat jakiegos innego fragmentu kodu. Musimy pomyslec co zrobic zeby nie bylo dwoch prawie identycznych metod. Nie jest to na ta chwile ale na kolejny PR prawdopodobnie